xdsl/interpreters/pdl_interp.py

```diff
@@ -34,6 +35,14 @@ class PDLInterpFunctions(InterpreterFunctions):
 
     _rewriter: PatternRewriter | None = field(default=None)
 
+    native_constraints: ClassVar[
+        dict[str, Callable[..., tuple[bool, tuple[Any, ...]]]]
+    ] = {}
+    """
+    The functions that can be used in `pdl_interp.apply_constraint`. Note that we do
+    not verify that the functions are used with the correct types.
+    """
```

Comment:
Is having this as a class var the done thing here? I'm not sure why but having a mutable class variable seems strange to me?

Reply:
I'd copied the `ClassVar` from `PDLMatcher`. I can remove it, I'm not even sure what it really does.

Reply:
Yep should definitely not be a ClassVar